chain/network/src/peer_manager/tests/tier2.rs

```diff
@@ -133,6 +133,43 @@ async fn test_reconnect_after_restart_outbound_side() {
     pm0.wait_for_direct_connection(id1.clone()).await;
 }
 
+#[tokio::test]
+async fn test_skip_reconnect_after_restart_outbound_side() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let pm0_db = TestDB::new();
+    let mut pm0_cfg = chain.make_config(rng);
+    pm0_cfg.connect_to_reliable_peers_on_startup = false;
+
+    let pm0 = start_pm(clock.clock(), pm0_db.clone(), pm0_cfg.clone(), chain.clone()).await;
+    let pm1 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+
+    let id1 = pm1.cfg.node_id();
+
+    tracing::info!(target:"test", "connect pm0 to pm1");
+    pm0.connect_to(&pm1.peer_info(), tcp::Tier::T2).await;
+    clock.advance(STORED_CONNECTIONS_MIN_DURATION);
+
+    tracing::info!(target:"test", "check that pm0 stores the outbound connection to pm1");
+    pm0.update_connection_store(&clock.clock()).await;
+    check_recent_outbound_connections(&pm0, vec![id1.clone()]).await;
+
+    tracing::info!(target:"test", "drop pm0 and start it again with the same db");
+    drop(pm0);
+    let pm0 = start_pm(clock.clock(), pm0_db.clone(), pm0_cfg.clone(), chain.clone()).await;
+
+    tracing::info!(target:"test", "check that pm0 has pm1 as a recent connection");
+    check_recent_outbound_connections(&pm0, vec![id1.clone()]).await;
+
+    // this will error if there's already a connection
+    tracing::info!(target:"test", "try connecting pm0 to pm1");
+    pm0.connect_to(&pm1.peer_info(), tcp::Tier::T2).await;
```

Comment:
That's true, unless connect_to() happens before pm0 tries to connect on its own.
Please check whether it is feasible to simply await completion of initial connections bootstrapping somehow (perhaps add an event for that). If not, add a TODO to properly await pm0 bootstrapping to complete and then check the connection pool directly.

Reply:
Since the reconnect attempts are async, checking the connection pool after peer manager starts is still timing-sensitive. Instead, I've added another event ReconnectLoopSpawned which is reported by the peer manager as soon as it decides to reconnect.